src/types.jl

```diff
@@ -677,34 +689,44 @@ element type $T is $bitstring type representing $(2^(8*sizeof(T))) values from $
   See the READMEs for FixedPointNumbers and ColorTypes for more information."""))
 end
 
-function throw_colorerror(::Type{N0f8}, values::Tuple{Vararg{Integer}})
+function throw_colorerror(::Type{C}, values::Tuple{Vararg{Integer}}) where C<:Colorant{N0f8}
     # Let's try to read the user's mind
-    if all(x->0<=x<=255, values)
-        if length(values) == 1
-            vstr = "$(values[1]) is an integer"
-            Tstr = "Gray"
-        else
-            vstr = "$values are integers"
-            if length(values) == 2
-                Tstr = "AGray"
-            elseif length(values) == 3
-                Tstr = "RGB"
-            else
-                Tstr = "RGBA"
-            end
-        end
-        args = join(map(v->"$v/255", values), ',')
-        throw(ArgumentError("""
+    all(x->0<=x<=255, values) || throw_colorerror_(N0f8, values)
+    if length(values) == 1
+        vstr = "$(values[1]) is an integer"
+    else
+        vstr = "$values are integers"
+    end
+    Cstr = colorant_string_with_eltype(C)
+    args = join(map(v->"$v/255", values), ',')
+    throw(ArgumentError("""
 $vstr in the range 0-255, but integer inputs are encoded with the N0f8
   type, an 8-bit type representing 256 discrete values between 0 and 1.
-  Consider dividing your input values by 255, for example: $Tstr{N0f8}($args)
+  Consider dividing your input values by 255, for example: $Cstr($args)
   Or use `reinterpret(N0f8, x)` if `x` is a `UInt8`.
   See the READMEs for FixedPointNumbers and ColorTypes for more information."""))
+end
+function throw_colorerror(::Type{C},
+                          values::Tuple{Vararg{Integer}}) where C<:Union{RGB24,ARGB32,Gray24,AGray32}
+    # For consistencies, Some sets of `UInt32` inputs are valid for the
+    # constructors of these non-parametric colors,
+    # e.g. `RGB24(UInt32(1), UInt32(0), UInt32(0))` is valid.
+    # Therefore, this error should be thrown after the range checking.
+    tripled(t) = t[1] isa UInt32 && t[1] === t[2] && t[2] === t[3]
```

Comment:
What's the intent behind checking that this is a grayscale represented with three components? What does that have to do with users providing bit-pattern inputs?

Reply:
Currently, the 1-arg constructors are provided via `convert`. 
https://github.com/JuliaGraphics/ColorTypes.jl/blob/5c589fa7903f65abdc71ac7f453c233f79bec54d/src/conversions.jl#L121-L123
https://github.com/JuliaGraphics/ColorTypes.jl/blob/5c589fa7903f65abdc71ac7f453c233f79bec54d/src/conversions.jl#L102-L103

Unless we change this flow or analyze the call stack, it is indistinguishable whether the user called the 1-arg or 3-arg constructor. So the `tripled()` is a heuristic and imperfect method.

```julia
julia> c = 0xC0FFEE;

julia> RGB24(c)
ERROR: ArgumentError: RGB24 cannot be constructed or converted directly from a UInt32 input as a bit pattern.
  Use `reinterpret(RGB24, 0x00c0ffee)` instead.

julia> RGB24(oftype.(c,(0x12, 0x34, 0x56))...) # I don't think this is the case, but use of `reinterpret` is a wrong advice.
ERROR: ArgumentError: element type FixedPointNumbers.Normed{UInt8,8} is an 8-bit type representing 256 values from 0.0 to 1.0,
  but the values (0x00000012, 0x00000034, 0x00000056) do not lie within this range.
  See the READMEs for FixedPointNumbers and ColorTypes for more information.
```

FYI, `ARGB32` is not supported for now due to https://github.com/JuliaGraphics/ColorTypes.jl/pull/183#issuecomment-616958191.

Reply:
Got it. Perhaps a comment in the source would help.

And yes, we want to avoid suggesting `reinterpret` when people explicitly provide the 3-arg version.

But as you hint, wouldn't a more reliable approach be to have `convert(::Type{RGB24}, x::Real)` do the checking and have *it* be the only path to suggest `reinterpret`?

Reply:
> But as you hint, wouldn't a more reliable approach be to have `convert(::Type{RGB24}, x::Real)` do the checking and have *it* be the only path to suggest `reinterpret`?

Yes. But the mechanism of construction and conversion is somewhat chaotic. :dizzy_face: That is one of the motivations of PR #197. 

I don't want this PR to be too "heavy", so I'll just add the comment in the source.